### ISSUE TITLE
No Firebase App '[DEFAULT]' has been created

### ISSUE THREAD

**Issue by wstidolph** [javebratt/debtTracker#2]
From a clean install of debtTracker, and with Chrome DevTools emptying the LocalStorage so no leftovers, I get an error when using the functions which use the firebase SDK directly (rather than going through AngularFire). For example, let's trigger AuthData#resetPassword: 

- ionic serve starts app, browser opens on "Start Creating Your Bills"
- open Chrome DevTools, verify no errors in console, nothing in LocalSTorage (delete firebase keys if present)
- click on ALREADY HAVE AN ACCOUNT button (no errors in console) get to LOGIN screen
- click on FORGOT MY PASSWORD, enter email, click on RESET YOUR PASSWORD
error in console:
```
error_handler.js:46 EXCEPTION: Error in ./ResetPasswordPage class ResetPasswordPage - inline template:22:40 caused by: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp().
ORIGINAL EXCEPTION: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp().
```

I suspect this is same problem as discussed in angular/angularfire2#556 so I tried injecting FirebaseApp into AuthData for access to the underlying firebase app:
```
import {Injectable, Inject} from '@angular/core';
import {
  AngularFire,
  AuthProviders,
  AuthMethods, FirebaseApp
} from 'angularfire2';
// ...

 constructor(public af: AngularFire, @Inject(FirebaseApp) firebaseApp: FirebaseApp) {
    this.firebaseApp = firebaseApp;

// ...

  resetPassword(email: string): any {
    return this.firebaseApp.auth().sendPasswordResetEmail(email);
  }
```
which my IDE is happy with, but it gets a transpilation error on attempting to run the app:

```
[13:30:40]  typescript: C:/Users/wayne/Projects/javebratt/debtTracker/src/providers/auth-data.ts, line: 12 
            Cannot find name 'FirebaseApp'.

      L11:  [13:30:40]  typescript: C:/Users/wayne/Projects/javebratt/debtTracker/src/providers/auth-data.ts, line: 15 export
 class AuthData {
      L12:    firebaseApp:FirebaseApp;

            Cannot find name 'FirebaseApp'.

```
Context: using Windows 10, clean clone of debtTracker
```
commit cdea84839662ed4384946634b99012f345b7aee9
Author: Jorge Vergara <javebratt@users.noreply.github.com>
Date:   Thu Oct 20 16:37:19 2016 -0500

C:\Users\wayne\Projects\javebratt\debtTracker>npm ls firebase
ionic-hello-world@ C:\Users\wayne\Projects\javebratt\debtTracker
+-- angularfire2@2.0.0-beta.5
| `-- firebase@3.3.0
`-- firebase@3.5.2
```

**Comment by javebratt** [comment]
Hey @wstidolph this started happening to me yesterday with the newest Firebase update, don't understand why, but if you go to the `auth-data.ts` provider and remove the `import firebase from 'firebase';` line it will get the correct reference when you do `firebase.auth().` and run.

Let me know if it works for you so I can close the issue, if it doesn't add your `ionic info` when you reply.

**Comment by wstidolph** [comment]
Sorry, I should have mentioned that I had already commented out that import line, before I posted this issue. Did you add in the FirebaseApp injection to make your code work?? I can't get that to work, I get:

```
[14:33:42]  Error: Error at C:/Users/wayne/Projects/javebratt/debtTracker/.tmp/providers/auth-data.ts:13:72 
[14:33:42]  Cannot find name 'FirebaseApp'. 
[14:33:42]  Error at C:/Users/wayne/Projects/javebratt/debtTracker/.tmp/providers/auth-data.ts:13:72 
[14:33:42]  Parameter 'firebaseApp' of constructor from exported class has or is using private name 'FirebaseApp'. 
```

My ionic info is:
```
Cordova CLI: 6.3.1
Gulp version:  CLI version 1.2.2
Gulp local:
Ionic Framework Version: 2.0.0-rc.1
Ionic CLI Version: 2.1.1
Ionic App Lib Version: 2.1.1
Ionic App Scripts Version: 0.0.37
OS:
Node Version: v6.7.0
```

**Comment by javebratt** [comment]
No, literally all I did was comment out the firebase import line and it worked.

Steps I followed:

* I cloned the repo into a new folder.

* I cd into it and ran `npm install`

* I ran `npm install --save @ionic/app-scripts@beta` since they released an update today.

* I ran `ionic serve`

* Got the issue.

* Went to `auth-data.ts` and removed the `import firebase from 'firebase';` line.

* Refreshed the page and then both `resetPassword()`, `link()` and `signOut()` worked fine, those are the ones using the Firebase JS SDK inside the `AuthData` provider.
